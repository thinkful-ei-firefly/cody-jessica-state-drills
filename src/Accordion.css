ul{
  display: flex;
  flex-direction: column;
  align-content: space-between;
  list-style: none;
}

button{
  padding: 10px;
  background-color: lightcoral;
  font-size: 15px;
  margin: 10px;
}

p{
  border: 2px solid darkgray;
  padding: 10px;
}