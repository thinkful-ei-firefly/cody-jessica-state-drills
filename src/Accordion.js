import React from 'react';

export default class Accordion extends React.Component {
  constructor(props) {
    super(props);
    this.state = { showIndex: null };
  }

  getSections() {
    const sectionList = this.props.sections;
    const jdxSectionList = [];

    sectionList.forEach((section, index) => {
      jdxSectionList.push(
        <div key={index}>
          <button
            
            onClick={() => this.setState({ showIndex: index })}
          >
            {section.title}
          </button>
          {(this.state.showIndex === index) && (<p>{section.content}</p>)}
        </div>


      )
    });
    // return jdxSectionList
    return jdxSectionList;
  }

  render() {
    return (
      <main className='App'>
        <ul>
          {this.getSections()}

        </ul>
      </main>
    );
  }
}

