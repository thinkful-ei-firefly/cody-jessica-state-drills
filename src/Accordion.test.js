import React from 'react';
import ReactDOM from 'react-dom';
import Renderer from 'react-test-renderer'
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe('<Accordion sections={sections}/>', () =>{
  it('Renders an empty list when no components are defined', () => {
    const tree = Renderer.create(<Accordion sections={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders no sections as active by default', () => {
    const tree = Renderer.create(<Accordion sections={sections} />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders a <p> element below the button that was clicked that displays content', () =>{
    const wrapper = shallow(<Accordion sections={sections}/>);
    wrapper.find('button').at(0).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('renders the last section when multiple sections have been clicked', () =>{
    const wrapper = shallow(<Accordion sections={sections}/>);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(2).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    wrapper.find('button').at(2).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})