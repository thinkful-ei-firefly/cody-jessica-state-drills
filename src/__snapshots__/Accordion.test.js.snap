// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion sections={sections}/> Renders <Accodion /> Correctly 1`] = `
<main
  className="App"
>
  <ul />
</main>
`;

exports[`<Accordion sections={sections}/> Renders <Accordion /> Correctly 1`] = `
<main
  className="App"
>
  <ul />
</main>
`;

exports[`<Accordion sections={sections}/> Renders an empty list when no components are defined 1`] = `
<main
  className="App"
>
  <ul />
</main>
`;

exports[`<Accordion sections={sections}/> renders a <p> element below the button that was clicked that displays content 1`] = `
<main
  className="App"
>
  <ul>
    <div
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </div>
    <div
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
    </div>
    <div
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
    </div>
  </ul>
</main>
`;

exports[`<Accordion sections={sections}/> renders no sections as active by default 1`] = `
<main
  className="App"
>
  <ul>
    <div>
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
    </div>
  </ul>
</main>
`;
